src/Microsoft.Framework.BufferEntryCollection.Internal/BufferEntryCollection.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Framework.Internal
+{
+    /// <summary>
+    /// An ordered collection of strings with an enumerator that iterates over it.
+    /// </summary>
+    internal class BufferEntryCollection : IEnumerable<string>
```

Comment:
@troydai the shared-source projects are a special case.  everything is `internal` because otherwise any assembly that depends on (for example) both Razor and MVC would see two versions of the classes.  and the classes in this repo are in the namespace we chose for it.